### PR TITLE
[9.0][kpi] FIX #1093

### LINKED ISSUE #1093: [9.0][kpi] Error impossible to fix without hack if Range value is Python code and formula is throws an error
If a range value is set to `Python` and code throws an error, this error gets thrown every time the record gets accessed, so it's impossible to fix it and/or delete it.

Errors should be catched and handled, allowing the user to fix it.
I'll push a fix as soon as I have the time for it, but wanted to report it in case I don't.

```
Odoo Server Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/openerp/http.py", line 650, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/lib/python2.7/dist-packages/openerp/http.py", line 687, in dispatch
    result = self._call_function(**self.params)
  File "/usr/lib/python2.7/dist-packages/openerp/http.py", line 323, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/service/model.py", line 118, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/http.py", line 316, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/lib/python2.7/dist-packages/openerp/http.py", line 966, in __call__
    return self.method(*args, **kw)
  File "/usr/lib/python2.7/dist-packages/openerp/http.py", line 516, in response_wrap
    response = f(*args, **kw)
  File "/usr/lib/python2.7/dist-packages/openerp/addons/web/controllers/main.py", line 835, in search_read
    return self.do_search_read(model, fields, offset, limit, domain, sort)
  File "/usr/lib/python2.7/dist-packages/openerp/addons/web/controllers/main.py", line 856, in do_search_read
    request.context)
  File "/usr/lib/python2.7/dist-packages/openerp/http.py", line 1071, in proxy
    result = meth(cr, request.uid, *args, **kw)
  File "/usr/lib/python2.7/dist-packages/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/models.py", line 5305, in search_read
    result = self.read(cr, uid, record_ids, fields, context=read_ctx)
  File "/usr/lib/python2.7/dist-packages/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/models.py", line 3208, in read
    result = BaseModel.read(records, fields, load=load)
  File "/usr/lib/python2.7/dist-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/models.py", line 3254, in read
    values[name] = field.convert_to_read(record[name], use_name_get)
  File "/usr/lib/python2.7/dist-packages/openerp/models.py", line 5769, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/usr/lib/python2.7/dist-packages/openerp/fields.py", line 831, in __get__
    self.determine_value(record)
  File "/usr/lib/python2.7/dist-packages/openerp/fields.py", line 940, in determine_value
    self.compute_value(recs)
  File "/usr/lib/python2.7/dist-packages/openerp/fields.py", line 895, in compute_value
    self._compute_value(records)
  File "/usr/lib/python2.7/dist-packages/openerp/fields.py", line 885, in _compute_value
    getattr(records, self.compute)()
  File "/usr/lib/python2.7/dist-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/opt/odoo/extra-addons/oca-server-tools/kpi/models/kpi_threshold_range.py", line 129, in _compute_max_value
    value = safe_eval(obj.max_code)
  File "/usr/lib/python2.7/dist-packages/openerp/tools/safe_eval.py", line 296, in safe_eval
    c = test_expr(expr, _SAFE_OPCODES, mode=mode)
  File "/usr/lib/python2.7/dist-packages/openerp/tools/safe_eval.py", line 153, in test_expr
    code_obj = compile(expr, "", mode)
  File "<string>", line 1
    >5
    ^
SyntaxError: invalid syntax
```

### PR DESCRIPTION
**Pull request by ivantodorovich**
This should FIX #1093 and added a few minor improvements

### CODE REVIEW

**Comment by ivantodorovich** [comment]
As I understand, commits are automatically squashed before merging, right?
Otherwise, please let me know so I squash them first.

File: kpi/__openerp__.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 {
     "name": "Key Performance Indicator",
-    "version": "9.0.1.0.0",
+    "version": "9.0.2.0.0",
```
Comment (max3903) [inline kpi/__openerp__.py]: @ivantodorovich A module upgrade is enough here, so this should be `9.0.1.1.0`.

**Review by max3903 — CHANGES_REQUESTED** [review]
Minor change otherwise LGTM

:+1:

**Review by max3903 — APPROVED** [review]

**Comment by ivantodorovich** [comment]
Hi @max3903 !

Do you know anyone else interested in these modules that might be able to review this?
I'd really like to keep working on improving all the mgmtsystem modules, as we use them intensively in our company. Would love to do it on the OCA repositories.

**Comment by max3903** [comment]
@ivantodorovich I don't know but 4 months is enough time.

**Comment by ivantodorovich** [comment]
Thanks!

**Pull request merged**